### DISCUSSION TITLE
Do laravel 8.x supports PHP 8.2?

### DISCUSSION THREAD

**Discussion by parallels999** [laravel/framework#45799] [Q&A] [answered]
Do laravel 8.x supports PHP 8.2?

**Comment by henzeb** [comment]
Well, i went through the releases searching for a version 8. 

https://github.com/laravel/framework/tree/v8.83.27

Looking at the composer requirements i'd say yes.

**Reply by parallels999** [reply]
If you look Laravel 7.x composer requirements, you'd say yes too??
https://github.com/laravel/framework/blob/v7.30.6/composer.json

 i went through the releases searching for a version 8, and the latest change for PHP was
>Avoid Passing null to parameter exception on PHP 8.1

**Reply by henzeb** [reply]
yes. "php": "^7.2.5|^8.0",

**Reply by parallels999** [reply]
> yes. "php": "^7.2.5|^8.0"

Not, because running test on Laravel 7.x with PHP >=8.1 fails,
maybe as it reached his end of live they forgot adding something like `"php": "^7.2.5|<=8.0`

**Reply by henzeb** [reply]
That's why "looking at ... I'd say yes". It was a quick look. It's the same thing as in the accepted answer which looks at the workflow file. The only difference is the workflow is a bit more conclusive. Running tests, as you did, even more. 

Tests can fail because something written in the test fails. Not necessarily because the tested feature would fail. I had tests fail on 7.4 because I tried to get class name from an anonymous class using '::class' in a test (not in the code I was testing). This can also be the other way around where a test is using an old function that was stripped in the current php version, while the code that's being tested actually works fine. 

Capping the last version to a specific php version is irrelevant as someone using it should upgrade to a newer version anyway and may lead to problems if someone got it to work in a newer php version as changes are usually non-breaking and backward incompatible changes are few. 

But still, if you want to be sure it runs on php 8.2, upgrade your laravel instance. Except in a few rare cases it should not be an issue.

**Comment by erikn69** [comment, marked as answer]
If you see the tests on 8.x branch it is not officially supported, it could work, but you have to try by your own risk, It would be better upgrade to the latest version.
https://github.com/laravel/framework/blob/d3ff07dbb343b7aaaef2cc623cc32516d8551a68/.github/workflows/tests.yml#L36-L39

- https://github.com/laravel/framework/commit/f310c74b4aa73336957de6b7830e0dbd2471a70e
